atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraMultinodeEteTest.java

```diff
@@ -0,0 +1,237 @@
+public class CassandraMultinodeEteTest {
+
+    private static final Gradle GRADLE_PREPARE_TASK =
+            Gradle.ensureTaskHasRun(":atlasdb-ete-test-utils:buildCassandraImage");
+
+    private static final int THRIFT_PORT_NUMBER_1 = 9160;
+    private static final int THRIFT_PORT_NUMBER_2 = 9161;
+    private static final int THRIFT_PORT_NUMBER_3 = 9162;
+
+    private static final int TIMELOCK_SERVER_PORT = 3828;
+
+    private static final String CONTAINER_LOGS_DIRECTORY = "container-logs/cassandra-multinode";
+
+    private static final List<String> CASSANDRA_NODES = ImmutableList.of("cassandra1", "cassandra2", "cassandra3");
+
+    private static final long MAX_CASSANDRA_NODE_DOWN_MILLIS = 30000;
+    private static final long MAX_CASSANDRA_NODES_RUNNING_MILLIS = 3000;
+
+    private static final DockerComposeRule CASSANDRA_DOCKER_SETUP = DockerComposeRule.builder()
+            .file("docker-compose.multiple-cassandra.yml")
+            .waitingForService("cassandra1", Container::areAllPortsOpen)
+            .waitingForService("cassandra2", Container::areAllPortsOpen)
+            .waitingForService("cassandra3", Container::areAllPortsOpen)
+            .waitingForService("ete1", toBeReady())
+            .saveLogsTo(CONTAINER_LOGS_DIRECTORY)
+            .build();
+
+
+    @ClassRule
+    public static final RuleChain CASSANDRA_DOCKER_SET_UP = RuleChain
+            .outerRule(GRADLE_PREPARE_TASK)
+            .around(CASSANDRA_DOCKER_SETUP);
+
+    @Test
+    public void shouldRunTransactionsWithAllCassandraNodesRunningWithoutUnacceptableDelay()
+            throws InterruptedException {
+        TodoResource clientToSingleNode = createClientFor(TodoResource.class, asPort("ete1"));
+
+        long transactionStartTime = System.currentTimeMillis();
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+        long transactionEndTime = System.currentTimeMillis();
+
+        long transactionTimeWithAllNodesRunning = transactionEndTime - transactionStartTime;
+
+        assertThat("transactionTimeWithAllNodesRunning",
+                transactionTimeWithAllNodesRunning,
+                is(lessThan(MAX_CASSANDRA_NODES_RUNNING_MILLIS)));
+    }
+
+    @Test
+    public void shouldRunTransactionsAfterCassandraNodeIsShutDownWithoutUnacceptableDelay()
+            throws InterruptedException {
+        TodoResource clientToSingleNode = createClientFor(TodoResource.class, asPort("ete1"));
+
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+
+        String cassandraNodeToKill = getRandomCassandraNodeToShutdown();
+        killCassandraContainer(cassandraNodeToKill);
+
+        long transactionStartTime = System.currentTimeMillis();
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+        long transactionEndTime = System.currentTimeMillis();
+
+        long transactionTimeAfterNodeIsKilled = transactionEndTime - transactionStartTime;
+
+        startCassandraContainer(cassandraNodeToKill);
```

Comment:
@gsheasby 
More cases:
- Gets work when the node to which the query is directed to is down (with large transaction time)
- Deletes work when the node to which the query is directed to is down (with large transaction time)
- Gets work when a node to which the query is not directed to is down (with small transaction time)
- Deletes work when a node to which the query is not directed to is down (with small transaction time)